packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -376,6 +380,46 @@ export class CallContext {
     });
   }
 
+  /* @conditional-compile-remove(local-recording-notification) */
+  public setCallRecordingInfos(
+    callId: string,
+    recordingInfosAdded: RecordingInfo[],
+    recordingInfosRemoved: RecordingInfo[]
+  ): void {
+    this.modifyState((draft: CallClientState) => {
+      const call = draft.calls[this._callIdHistory.latestCallId(callId)];
+      if (call) {
+        call.recording.activeRecordings = recordingInfosAdded;
+        call.recording.stoppedRecordings = recordingInfosRemoved;
+      }
+    });
+  }
+
+  /* @conditional-compile-remove(local-recording-notification) */
+  public setCallLocalRecordingActive(callId: string, isRecordingActive: boolean): void {
+    this.modifyState((draft: CallClientState) => {
+      const call = draft.calls[this._callIdHistory.latestCallId(callId)];
+      if (call) {
+        call.localRecording.isRecordingActive = isRecordingActive;
+      }
+    });
+  }
+
+  /* @conditional-compile-remove(local-recording-notification) */
+  public setCallLocalRecordingInfos(
+    callId: string,
+    localRecordingInfosAdded: LocalRecordingInfo[],
+    localRecordingInfosRemoved: LocalRecordingInfo[]
+  ): void {
+    this.modifyState((draft: CallClientState) => {
+      const call = draft.calls[this._callIdHistory.latestCallId(callId)];
+      if (call) {
+        call.localRecording.activeRecordings = localRecordingInfosAdded;
```

Comment:
Would this event only provide the latest active recordings added? Would we ever need to keep track of recordings from previous 'localRecordingsUpdated' events that are still active which would be lost here?

Reply:
This code will need a bit more refactoring. I was under the assumption that the event 'localRecordingsUpdated' returned the full list of added and removed recordings. However that is not the case.

I got more information today from the calling sdk engineer Karan MIshra.
He mentioned that the event for 'localRecordingsUpdated' only returns one instance of a recordingInfo.
This means that, the event will only ever return 1 instance of an Added OR Removed recordingInfo.

However, the `recordings` getter, not the event but another function from the feature, retrieves all ACTIVE recordings upon calling the getter. 

There is no collection of removed/stopped state recordings maintained by the calling sdk. Only for active recordings. 

Knowing this, we will need to cache and maintain a list of recordings in ether our state manager or in the compliance banner state manager used in the composites.

Reply:
I see. That is similar to spotlight. The spotlight feature object also has a getter for spotlighted participants and events that has the added and removed array.

For the stopped state recordings, I would probably manage the state in react-composites since we don't want to track dismissed time in stateful. I'm assuming that stopped state recordings are only for notifications.

Reply:
That's right, stopped state recordings would only be for notifications.
From your experience, what do you think is the best approach here?

1. Have a list of active recordings that always returns only the active ones. (then in the composite we can do a diff check against cached)

2. Have list of active recordings. A state param that returns a removed recording. Then in composite If it's a new removed recording we would display banner. If active recordings get any new entries then display banner.


3. Return just one list (a set) that contains all LocalRecordingInfo keyed by displayName with value being state. (started, stopped, none, etc)

Reply:
Actually looking back at the capabilities code, I added a prop [latestCapabilitiesChangeInfo](https://github.com/Azure/communication-ui-library/blob/main/packages/calling-stateful-client/src/CallClientState.ts#L150) in the stateful feature object. Then I would have access to the changes in capabilities from latest event of 'capabilitiesChanged' through a selector where the composite can track the changes using a hook like shown [here](https://github.com/Azure/communication-ui-library/blob/main/packages/react-composites/src/composites/CallComposite/CallComposite.tsx#L412)

In the same way maybe, you can do something similar and add a prop to LocalRecordingCallFeature (something like latestLocalRecordingInfo) and then composite can track the changes using a hook.

Reply:
Are we showing one notification that summarizes all active recordings and another for all stopped recordings? Or each active/stopped recording gets a notification?